lib/handler/redirect.c

```diff
@@ -101,6 +149,7 @@ h2o_redirect_handler_t *h2o_redirect_register(h2o_pathconf_t *pathconf, int inte
     self->super.on_req = on_req;
     self->internal = internal;
     self->status = status;
-    self->prefix = h2o_strdup(NULL, prefix, SIZE_MAX);
+    build_prefix_list(h2o_iovec_init(prefix, strlen(prefix)), &self->prefix_list);
```

Comment:
I think you can pass in a NULL-terminated string rather than an iovec, as `prefix` is the string.

Reply:
done in https://github.com/h2o/h2o/pull/2186/commits/7405d448a51c179ab5668139ff539e3edddfe1f8